types/react-calendar-heatmap/index.d.ts

```diff
@@ -0,0 +1,32 @@
+// Type definitions for react-calendar-heatmap v1.6.1
+// Project: https://github.com/patientslikeme/react-calendar-heatmap
+// Definitions by: Keisuke Kan <https://github.com/9renpoto>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.3
+
+/// <reference types="react" />
+
+declare module "react-calendar-heatmap" {
+    interface Props {
+        classForValue?: Function
+        endDate?: string | number | Date
+        gutterSize?: number
+        horizontal?: boolean
+        monthLabels?: string[]
+        numDays?: number
+        onClick?: Function
+        onMouseLeave?: Function, 
+        onMouseOver?: Function, 
+        showMonthLabels?: boolean
+        showOutOfRangeDays?: boolean
+        showWeekdayLabels?: boolean, 
+        startDate?: string | number | Date,
+        titleForValue?: Function,
+        tooltipDataAttrs?: Object | Function
```

Comment:
`Object | Function` is just `Object`.  Also, you don't want to use `Object` because anything is structurally assignable to it.

Consider using `object` (notice the lowercase) or `any` instead. The lint rules should help out here.